app/build.gradle

```diff
@@ -146,4 +145,5 @@ dependencies {
   testCompile 'org.assertj:assertj-core:1.7.0'
   testCompile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
   testCompile 'org.json:json:20160212'
+  testCompile 'com.google.dagger:dagger-compiler:2.0'
```

Comment:
Good call. I updated the build script.

Yes, this was what was causing the failure. I found this thread: https://bitbucket.org/hvisser/android-apt/issues/22/dagger-robolectric-gradle-apt-source-not